packages/shared/src/utils/validator/__tests__/validator.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { TFormErrorMessagesTypes } from '../../validation/form-error-messages-types';
+import { initFormErrorMessages } from '../../validation';
+import Validator, { template } from '../validator';
+
+describe('validator.ts', () => {
+    const amount_required_error = 'Amount is a required field.';
+    const should_be_valid_number_error = 'Should be a valid number.';
+    const only_99_characters_error = 'Only 99 characters, please.';
+    const min_value_error = 'The value must be equal or greater than 1';
+
+    beforeAll(() => {
+        initFormErrorMessages({ number: () => should_be_valid_number_error } as TFormErrorMessagesTypes);
+    });
+
+    describe('template', () => {
+        it('should return a string where variables in brackets are replaced with respective items from content array', () => {
+            expect(template('abc [_1] abc', ['2'])).toBe('abc 2 abc');
+            expect(template('[_1] [_2]', ['1', '2'])).toBe('1 2');
+            expect(template('[_1] [_1]', ['1'])).toBe('1 1');
+        });
+        it('should replace variables in brackets with items from content array only once, not twice', () => {
+            expect(template('[_1] [_2]', ['[_2]', 'abc'])).toBe('[_2] abc');
+            expect(template('[_1] [_2]', ['[_2]', 'abc'])).not.toBe('abc abc');
```

Comment:
Sorry, didn't get the idea. If returned value is `.toBe('[_2] abc')`, then why do we need to check same returned value that it is `.not.toBe('abc abc');`?

Reply:
@kate-deriv the second check is only presentational, it was meant to show what we don't want to get. i can remove it if its purpose is unclear here. :)